### DISCUSSION TITLE
Replace Laminas Code with Nikic's PHP-Parser

### DISCUSSION THREAD

**Discussion by simivar** [phpro/soap-client#359] [Ideas]
### Feature Request

Currently, Soap Client uses [Laminas Code](https://github.com/laminas/laminas-code) to generate code with `generate` commands. What do you think about switching it to [Nikic PHP-Parser](https://github.com/nikic/PHP-Parser)? 

There are several benefits but the main one is that PHP-Parser is used by many many more libraries (according to Github it's 922k vs 90k) like PHPStan and others and thanks to that all the newest features are usually supported really fast. 

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | yes
| BC Break    | yes

**Comment by fbecker-complex** [comment]
This would also fix https://github.com/phpro/soap-client/issues/355

**Comment by veewee** [comment]
This is not an easy-switch ... It implies that we have to rewrite the complete code generation part from scratch and deal with nikic/php-parser specific issues while working our way through it.


One of the questions I ask myself: what value do we actually gain from doing this massive destructive switch?
It is not that the new PHP features that got introduced in 8.0 or 8.1 are very useable in the generated soap code.
Even if it takes a couple of months to wait until it is implemented in laminas-code: I don't see it as a downside.

I rather spend that time working on some more blocking issues inside this package like advanced metadata parsing and improved code generation on additional XSD information.

The numbers of downloads you provide might also not reflect reality for various reasons:

- laminas got renamed from the zend framework namespace quite recently
- php-parser is frequently being used to traverse / alter code, meaning a lower part of the codebase is used for code generation.
- laminas-code is still well maintained IMO

Sure there might have been some breaking changes between v3 and v4, meaning we have to change stuff in this repo.
Those things are not specific to the laminas package, but might also occur in php-parser at some point in time.
It is expected behaviour of semantic versioning.


Are there any specific reasons why you want to switch code-generating package?
Is there something you are not able to do at this point?
